Add documentation for new opnsense integration

mtreinish opened the pull request:
**Description:**

This commit adds documentation for the opnsense device tracker which is
being added in home-assistant/home-assistant#26834.

**Pull request in home-assistant (if applicable):** home-assistant/home-assistant#26834

## Checklist:

- [X] Branch: `next` is for changes and new documentation that will go public with the next Home Assistant release. Fixes, changes and adjustments for the current release should be created against `current`.
- [X] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

mtreinish committed:
Add documentation for new opnsense component

This commit adds documentation for the opnsense device tracker which is
being added in home-assistant/home-assistant#26834.

klaasnicolaas commented on source/_components/opnsense.markdown:
```diff
@@ -0,0 +1,66 @@
+  - Hub
+  - Presence Detection
+ha_release: 0.100
+redirect_from:
```
Please remove the `redirect_from`

mtreinish replied:
Done in https://github.com/home-assistant/home-assistant.io/pull/10427/commits/dc933232ab654d79a27cc0a16f8ca13b16196d46

klaasnicolaas commented on source/_components/opnsense.markdown:
```diff
@@ -0,0 +1,66 @@
+ha_category:
+  - Hub
+  - Presence Detection
+ha_release: 0.100
```
```suggestion
ha_release: "0.100"
```

klaasnicolaas commented on source/_components/opnsense.markdown:
```diff
@@ -0,0 +1,66 @@
+  description: The url for the OPNsense api endpoint of your router
+  type: string
+  required: true
+  default: None
```
If there is no default in the python code, you can leave it away.

mtreinish replied:
Done in https://github.com/home-assistant/home-assistant.io/pull/10427/commits/dc933232ab654d79a27cc0a16f8ca13b16196d46

mtreinish committed:
Update source/_components/opnsense.markdown

Co-Authored-By: Klaas Schoute <klaas_schoute@hotmail.com>

mtreinish committed:
Fixes from review comments

klaasnicolaas commented on source/_components/opnsense.markdown:
```diff
@@ -0,0 +1,63 @@
+  description: The API key used to authenticate with your OPNsense API endpoint
+  type: string
+  required: true
+  default: None
```
Also here

mtreinish replied:
Oops, heh I guess I applied your earlier suggestion a bit too mechanically. Fixed in: https://github.com/home-assistant/home-assistant.io/pull/10427/commits/ddaab6b128d0a1c6376d13cebc22ec5cffa1eb22

klaasnicolaas commented on source/_components/opnsense.markdown:
```diff
@@ -0,0 +1,63 @@
+  description: The API secret used to authenticate with your OPNsense API endpoint
+  type: string
+  required: true
+  default: None
```
And here

mtreinish replied:
Fixed in: https://github.com/home-assistant/home-assistant.io/pull/10427/commits/ddaab6b128d0a1c6376d13cebc22ec5cffa1eb22

klaasnicolaas commented on source/_components/opnsense.markdown:
```diff
@@ -0,0 +1,63 @@
+  description: List of the OPNsense router's interfaces to use for tracking devices
+  type: list
+  required: false
+  default: None
```
And here 😉

mtreinish replied:
This one actually sets a default, just in the device tracker: https://github.com/home-assistant/home-assistant/pull/26834/files#diff-d7915733a6a8b679b376fc58c0b252caR14 I've updated this to reflect that in: https://github.com/home-assistant/home-assistant.io/pull/10427/commits/ddaab6b128d0a1c6376d13cebc22ec5cffa1eb22

mtreinish committed:
More updates from code review

The previous updates in dc933232ab654d79a27cc0a16f8ca13b16196d46 around
the defaults were applied a bit too mechanically and the same problem
existed elsewhere in the configuration section. This commit also updates
those to fix the same issue there.

frenck committed:
:pencil2: Tweaks

frenck committed:
Move documentation to new integrations location

frenck commented:
Rebased the PR onto the latest `next` branch, moved the Markdown file to the new location in the repository.

frenck reviewed (dismissed):
✅ Approved. Can be merged as soon as the parent PR gets merged.
⚠️ The `ha_release` needs an update before merging this one in.

mtreinish committed:
Remove LAN default for tracker_interfaces

Part of the review changes made to the PR adding this component was to
not set a default value for the tracker_interfaces to ["LAN"]. This
commit just updates the configuration section of the docs to reflect
this change.

frenck committed:
Update opnsense.markdown
